### ISSUE TITLE
🐛 Bug Report: MigrationLocked: Plugin 'catalog' startup failed; Migration table is already locked

### ISSUE THREAD

**Issue by yazidmissaoui** [backstage/backstage#26546]
### 📜 Description

Backstage fails to start due to a locked migration table for the catalog plugin. This error persists even after completely resetting the database and redeploying Backstage.

### 👍 Expected behavior

Backstage should start successfully, with the catalog plugin initializing without any migration lock issues.

### 👎 Actual Behavior with Screenshots

The deployment fails with the following error:
`MigrationLocked: Plugin 'catalog' startup failed; caused by Mi
![image](https://github.com/user-attachments/assets/807d18f3-1a38-4796-998b-4f1f41459ad7)
grationLocked: Migration table is already locked`

### 👟 Reproduction steps

1. Deploy Backstage using Helm with the following configuration: 
`backstage:
  image:
    registry: ghcr.io
    repository: backstage/backstage
    tag: latest
    pullPolicy: IfNotPresent
  replicaCount: 1
  resources:
    limits:
      cpu: 500m
      memory: 1024Mi
    requests:
      cpu: 100m
      memory: 256Mi
  podSecurityContext:
    runAsUser: 1000
    runAsGroup: 3000
    fsGroup: 2000
  securityContext:
    runAsNonRoot: true
    runAsUser: 1000
  appConfig:
    app:
      baseUrl: https://my_URL
    backend:
      baseUrl: https://my_URL
      database:
        client: pg
        connection:
          host: backstagedbserver.xxxxxxxxxxxxxxx.rds.amazonaws.com
          port: 5432
          user: postgres
          password: "mypwd"
    techdocs:
      builder: 'local'
      generator:
        runIn: 'local'
      publisher:
        type: 'local'
    catalog:
      locations: []
postgresql:
  enabled: false

features:
  enableTechdocs: true
global:
  serviceAccountName: backstage`

2. Wait for the deployment to complete
3. Check the logs of the Backstage pod
4. Observe the MigrationLocked error

### 📃 Provide the context for the Bug.

This issue is preventing the deployment of Backstage in our EKS Kubernetes cluster . We are unable to use any of the Backstage features due to this startup failure. We have tried : 

1. Resetting the database, 
2. Manually unlocking the migration table, 
3. Adjusting the configuration to skip migrations

**### but the issue persists.**

### 🖥️ Your Environment

Amazon Elastic Kubernetes Service
Kubernetes version: 1.30
PostgreSQL version: 16.3, Amazon RDS for PostgreSQL
Helm chart version: 1.9.6, https://artifacthub.io/packages/search?repo=backstage
Cloud Provider: AWS

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by freben** [comment]
Does this truly persist permanently, or is it a transient thing?

The way that knex migrations work, is that they reject attempts to run if some other node is currently running them. Normally the deployment system would retry deployment of these nodes some time later, and then they'd go through fine because the first node has completed the migration work.

**Comment by yazidmissaoui** [comment]
> Does this truly persist permanently, or is it a transient thing?
> 
> The way that knex migrations work, is that they reject attempts to run if some other node is currently running them. Normally the deployment system would retry deployment of these nodes some time later, and then they'd go through fine because the first node has completed the migration work.

It persists permanently. We have tried :

- Resetting the database,
- Manually unlocking the migration table,
- Adjusting the configuration to skip migrations
### but the issue persists.

**Comment by ioboi** [comment]
Hey @yazidmissaoui 

I experienced the same. For me the error persisted because the application will fail if the `guest-provider` is enabled in `production` ([see this comment by @freben ](https://github.com/backstage/backstage/issues/24284#issuecomment-2194902767)).

I was not able to spot the error in the pod logs, but removing the `guest-provider` fixed my deployment.

From the official [docs](https://backstage.io/docs/auth/guest/provider):

> This provider should only ever be enabled for `development`. To prevent unauthorized access to your data, this package is explicitly disabled for non-development environments.

With the release of https://github.com/backstage/backstage/pull/24188 it may be helpful to rephrase this to:

> This provider should only ever be enabled for `development`. To prevent unauthorized access to your data, this package errors in production and will prevent startup.

What do you think @freben. I can create a PR or follow up issue for this.
